termqt/terminal_widget.py

```diff
@@ -244,7 +249,36 @@ def set_font(self, font: QFont = None):
 
         self.row_len = int(self._width / self.char_width)
         self.col_len = int(self._height / self.line_height)
+    def set_font_qt6(self, font: QFont = None):
 
```

Comment:
May I know what is the reason for creating a new `set_font` function for qt6? I think most of this function is identical to the old one. Is there a way to reduce the duplication of the code here?

Reply:
`QFontDatabase` behaves slightly differently in `Qt5` and `Qt6`. Some parts of the legacy code that were already deprecated in `Qt5` were actually removed in `Qt6`.

Example of why two methods are needed:

If you run PyQt6 or PySide6, you will encounter a TypeError here:

In PyQt6, using `qfd = QFontDatabase()`:
```python
line 222, in set_font
    qfd = QFontDatabase()
          ^^^^^^^^^^^^^^^
TypeError: QFontDatabase(a0: QFontDatabase): not enough arguments
```

However, if `qfd = QFontDatabase()` is not preallocated in Qt5 versions, you will also encounter a TypeError:

In PyQt5, without `qfd = QFontDatabase()`:
```python
line 260, in set_font
    elif "Menlo" in QFontDatabase.families():
                    ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: families(self, writingSystem: QFontDatabase.WritingSystem = QFontDatabase.Any): first argument of unbound method must have type 'QFontDatabase'
```

However, you are right that methods can be consolidated better to avoid code repetition. I made a new commit to address this 9e5d807bc9c1faf17eaa513be0eb0c35fdbabd23

Reply:
Ok, make sense, I rewrote the methods.